### ISSUE TITLE
Need to be able to freeze the underlying property on the Window object

### ISSUE THREAD

**Issue by erights** [domenic/window-proxy-spec#3]
On the latest FF Nightly where this is deployed, visiting https://rawgit.com/google/caja/master/src/com/google/caja/ses/explicit.html shows diagnostics like

    Global Array cannot be made readonly: TypeError: Not allowed to define a non-configurable property on the WindowProxy object. New symptom(6) is not SES-safe.

for all of SES's whitelisted globals. There are two bugs here:

  1. The minor problem here is that we need to upgrade SES so that it copes with this new WindowProxy behavior. Since I had pushed so hard for this change to WindowProxy, I'm embarrassed that I haven't already done this.
  1. The major problem here is that it must be possible to make the underlying data property on the Window object non-configurable non-writable. Unless SES can lock it down, it must reject that platform as insecurable, as it currently does with Nightly.

Looking back over the original thread, there were various semantics proposed. All proposals that were correct -- that maintain the invariants -- must never allow the WindowProxy to claim that the property is non-config on the WindowProxy. But some of these proposals did first act on an attempt to defineProperty to non-config, non-write by *first* forwarding the operation to the Window, and only after rejecting the operation with an exception. This would still enable one (e.g., SES) to lock down global properties on the underlying Window object.

Testing on Nightly as well as the current draft spec text as of this writing, which does correspond to another semantics proposed on that thread (and which I did not object to at the time) does not seem to provide any way to lock down these global properties, making it impossible for SES to secure the platform.

**Comment by domenic** [comment]
Can you outline the exact change to the current spec text? I think I see it, but want to be sure.

**Comment by erights** [comment]
I just realize I may have misunderstood something. Will respond again after doing some tests.

**Comment by erights** [comment]
Once I reread the thread at https://esdiscuss.org/topic/figuring-out-the-behavior-of-windowproxy-in-the-face-of-non-configurable-properties more carefully, I realized how the spec as is does allow me to lock down a property. Testing on FF confirms that this technique works.

It seems to me now that the spec is correct and the FF implementation is at least mostly correct. Closing.
None of this will be a problem for SES once I upgrade SES.
